plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteTableProperties.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ignite;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.trino.plugin.jdbc.TablePropertiesProvider;
+import io.trino.spi.session.PropertyMetadata;
+import io.trino.spi.type.ArrayType;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.plugin.ignite.IgniteTemplateType.PARTITIONED;
+import static io.trino.plugin.ignite.IgniteWriteSyncMode.FULL_SYNC;
+import static io.trino.spi.session.PropertyMetadata.enumProperty;
+import static io.trino.spi.session.PropertyMetadata.integerProperty;
+import static io.trino.spi.session.PropertyMetadata.stringProperty;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class IgniteTableProperties
+        implements TablePropertiesProvider
+{
+    private final List<PropertyMetadata<?>> tableProperties;
+
+    public static final String PRIMARY_KEY_PROPERTY = "primary_key";
+    public static final String BACK_UPS_PROPERTY = "backups";
+    public static final String AFFINITY_KEY_PROPERTY = "affinity_key";
+    public static final String TEMPLATE_PROPERTY = "template";
+    public static final String WRITE_SYNCHRONIZATION_MODE_PROPERTY = "write_synchronization_mode";
+    public static final String CACHE_GROUP_PROPERTY = "cache_group";
+    public static final String CACHE_NAME_PROPERTY = "cache_name";
+    public static final String DATA_REGION_PROPERTY = "data_region";
```

Comment:
Do you have tests where you create tables with all properties used?

Reply:
Added a test in ``TestIgniteConnector#testCreateTable``